setup.sh

```diff
@@ -0,0 +1,13 @@
+pip install PyYAML
```

Comment:
When is this setup.sh file used? I don't see it in the yaml config...
I was thinking maybe we should instead we pipenv which uses the already existing Pipfile to manage dependencies

Reply:
I was using this setup.sh to make sure all the required packages (especially the pre-commit setting) are setup. Tbh I am not sure how to do that in a Pipfile but I have the instinct that that would be better. 

Would you be able to just update this in this branch? Once you update you could just approve this PR.

Reply:
no problem